Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -625,64 +1025,55 @@ public DensityOutput getSimpleDensity(double[] point) {
      *                              missing values.
      * @return A point with the missing values imputed.
      */
-    public double[] imputeMissingValues(double[] point, int numberOfMissingValues, int[] missingIndexes) {
-        checkArgument(numberOfMissingValues >= 0, "numberOfMissingValues must be greater than or equal to 0");
-
-        // We check this condition in traverseForest, but we need to check it here s
-        // wellin case we need to copy the
-        // point in the next block
-        checkNotNull(point, "point must not be null");
+    public ArrayList<double[]> getConditionalCentralField(double[] point, int numberOfMissingValues,
```

Comment:
the return data type should be abstract. and the comment needs update.

Reply:
issue 238. The return datatype is double[] for each tree and we just present a list. Note though that this is not the precision issue -- all input/output is double/double[]. The float[]/double[] is internal.